test/functional/sqlsrv/sqlsrv_ae_datetimes_as_strings.phpt

```diff
@@ -0,0 +1,595 @@
+--TEST--
+Test various date and time types with AE and ReturnDatesAsStrings set to true
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+// Check for expected errors. These are expected in cases where the dates and
+// times do not comply with ODBC standards.
+function ExecutePreparedStmt($stmt)
+{
+    if ($stmt == false) {
+        fatalError("Failure in sqlsrv_prepare");
+    } else {
+        $r = sqlsrv_execute($stmt);
+        if ($r == false) {
+            $errors = sqlsrv_errors();
+            if ($errors[0]['SQLSTATE'] != '22018' and
+                $errors[0]['SQLSTATE'] != '22008' and
+                $errors[0]['SQLSTATE'] != '22007' and
+                $errors[0]['SQLSTATE'] != '07006') {
+                print_r($errors);
+                fatalError("Unexpected error");
+            }
+        }
+    }
+}
+
+// Compare dates retrieved from the database with the date used for testing.
+// $testingDate is an array of strings like '2002-01-31 23:59:59.049876' and
+// $retrieved_date is the string date of the format 'Y-m-d H:i:s.u', which 
+// the format returned by the date_format() calls below or by retrieval as 
+// strings, unless $datetimetype is date or time. In that case, if 
+// $returnStrings is false and $datetimetype is 'date', the time defaults to
+// 00:00:00.0000, and if $datetimetype is 'time' and $returnStrings is false,
+// the date defaults to the current date. If however $returnStrings is true
+// and $datetimetype is 'date', $retrieved_date is only a date,
+// and if $datetimetype is 'time' and $returnStrings is true, $retrieved_date
+// only a time.
+// The concatenations involving zeroes below are to make direct string
+// comparisons feasible. Also, because PHP maxes out at microsecond precision
+// and SQL Server maxes out at 0.1 microsecond precision, the more precise 
+// types require an extra 0 for some comparisons.
+function CompareDateTime($datetimetype, $returnStrings, &$testingDate, $retrievedDate)
+{
```

Comment:
Suggestion: refactor this method into one for date, one for time or one for combination?